### PR TITLE
Feat/60 apply to postings

### LINKED ISSUE #55: Allow users to apply to job postings
### Describe why the project needs this task
Main interaction for the front-end with the job applications API: Users are able to apply to a job posted on our job board.

### Related user stories
* S8

### Describe the solution to the problem
Add some basic front-end handling when clicking on a job posting. This will include presenting a form to the user, after which they will make a query to the job applications API.

### Describe alternatives you've considered (for team discussion)
N/A

### Additional context
N/A

### PR DESCRIPTION
**Pull request by erickzhao**
Closes #55, #60

### CODE REVIEW

File: src/Features/Postings/ViewPosting.tsx [hunk 1]
```diff
@@ -73,9 +63,6 @@ const ViewPosting: React.FunctionComponent<
       .then(response => {
         if ("status" in response.data) {
           toast.error(response.data.status);
```
Comment (AndreiUngur) [inline src/Features/Postings/ViewPosting.tsx:65]: :+1: Good stuff, small nit: you can probably even change line 64 from `if ("status" in response.data)` to `if (response.data.status)` but it's really not a big deal, as long as it works!

**Comment by AndreiUngur** [comment]
Good work in here, thanks for the PR!

File: src/Features/Postings/Postings.tsx [hunk 2]
```diff
@@ -37,6 +27,16 @@ const styles = (theme: Theme) =>
   });
 
 const Postings: React.FunctionComponent<WithStyles> = ({ classes }) => {
+  const [postings, setPostings] = useState<IPosting2[]>([]);
+
+  useEffect(() => {
+    const fetchData = async () => {
```
Comment (alexboii) [inline src/Features/Postings/Postings.tsx:33]: own method if possible

**Review by alexboii — APPROVED** [review]
lgtm, only one minor comment

**Pull request merged**